inventory/client.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2022 Dell Inc, or its subsidiaries.
+
+// Package inventory implements the go library for OPI to be used to query inventory
+package inventory
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	grpcOpi "github.com/opiproject/godpu/grpc"
+	pb "github.com/opiproject/opi-api/common/v1/gen/go"
+	"google.golang.org/grpc"
+)
+
+// PbInvClientGetter defines the function type used to retrieve an inventory protobuf client
+type PbInvClientGetter func(c grpc.ClientConnInterface) pb.InventorySvcClient
+
+type invClientImpl struct {
+	getInvClient PbInvClientGetter
+	grpcOpi.Connector
+}
+
+// InvClient is an interface for querying inventory data from an OPI server
+type InvClient interface {
+	Get(ctx context.Context) (*pb.InventoryGetResponse, error)
+}
+
+// New creates an inventory client for use with OPI server at the given address
+func New(addr string) (InvClient, error) {
+	c, err := grpcOpi.New(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	// Default is to use the OPI grpc client and the pb client generated from the protobuf spec
+	return NewWithArgs(c, pb.NewInventorySvcClient)
+}
+
+// NewWithArgs creates an inventory client for use with OPI server using the given gRPC client and the given function for
+// retrieving an inventory protobuf client
+func NewWithArgs(c grpcOpi.Connector, getter PbInvClientGetter) (InvClient, error) {
+	if c == nil {
+		return nil, errors.New("grpc connector is nil")
+	}
+
+	if getter == nil {
+		return nil, errors.New("protobuf client getter is nil")
+	}
+
+	return invClientImpl{
+		getInvClient: getter,
+		Connector:    c,
+	}, nil
+}
+
+// Get returns inventory information an OPI server
+func (c invClientImpl) Get(ctx context.Context) (*pb.InventoryGetResponse, error) {
+	conn, closer, err := c.NewConn()
+	if err != nil {
+		log.Printf("error creating connection: %s\n", err)
+		return nil, err
+	}
+	defer closer()
+
+	client := c.getInvClient(conn)
+
+	data, err := client.InventoryGet(ctx, &pb.InventoryGetRequest{})
+	if err != nil {
+		log.Printf("error getting inventory: %s\n", err)
+		return nil, err
+	}
+
+	return data, nil
```

Comment:
I'm always wondering if better design choice when using protobufs is to create a translation struct and not return the same object to user... in case PB changes, for example... maybe we don't want client to know about it... and still use the same function as before... 
just a thought... not a blocker for this PR...

Reply:
I thought about this during implementation, but wasn't sure what the best practice should be. I think a translation layer would be a good idea though.

Reply:
We can create issue to track this future work